src/manager.rs

```diff
@@ -135,7 +146,8 @@ impl<Env, T> ScriptManager<Env, T>
                                         &[&id.to_string()]));
             },
             (true, false) => {
-                try!(self.start_script(id, &source));
+                let owner = try!(self.get_owner(&id));
```

Comment:
Mmmh... so we are opening and closing the connection twice for a single change? That looks wasteful/fragile.